From: Dmitry Timoshkov <dmitry@baikal.ru>
Subject: [PATCH 3/3] msxml3: Don't force UTF-8 when saving XML document.
Message-Id: <20210319182532.0d2062c4790573af9d3385d4@baikal.ru>
Date: Fri, 19 Mar 2021 18:25:32 +0300

This is the only place where xmlSaveToIO() is forced to use UTF-8 for an
output document, other places specify NULL for the default encoding.

This doesn't completely fix the saved XML contents, but at least XML document
has proper encoding now.

Signed-off-by: Dmitry Timoshkov <dmitry@baikal.ru>
---
 dlls/msxml3/domdoc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/msxml3/domdoc.c b/dlls/msxml3/domdoc.c
index 11111111111..11111111111 100644
--- a/dlls/msxml3/domdoc.c
+++ b/dlls/msxml3/domdoc.c
@@ -1440,7 +1440,7 @@ static HRESULT WINAPI domdoc_get_xml(
         return E_OUTOFMEMORY;
 
     options = XML_SAVE_FORMAT | XML_SAVE_NO_DECL;
-    ctxt = xmlSaveToIO(domdoc_get_xml_writecallback, NULL, buf, "UTF-8", options);
+    ctxt = xmlSaveToIO(domdoc_get_xml_writecallback, NULL, buf, NULL, options);
 
     if(!ctxt)
     {

-- 
0.0.0

